### PR TITLE
Kudex Finance && Kswap Finance && Koffee Swap && Cycle Finance && Zabu Fiance TVL Adapters And "Updating of portedTokens.js file"

### PR DESCRIPTION
**Pull request by Mohammed18062020**
##### Twitter Link:
Kudex -> https://twitter.com/kudexkcc
Kswap -> https://twitter.com/kswap_finance
Koffee Swap -> https://twitter.com/koffeeswap
Cycle Finance -> https://twitter.com/CycleProtocol
Zabu Finance -> https://twitter.com/zabufinance

##### Website Link:
Kudex -> https://kudex.finance/
Kswap -> https://kswap.finance/
Koffee Swap -> https://koffeeswap.exchange/#/
Cycle Finance -> https://www.cycle.finance/
Zabu Finance -> https://zabu.finance/#/

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
Kudex -> 
image

Kswap -> 
image

Koffee Swap -> 
image

Cycle Finance ->
image
Zabu Finance -> 
image

##### Current TVL:
Kudex -> $909,622.13
Kswap -> $70.42 M
Koffee Swap -> $2,443,981.45
Cycle Finance -> $22,014,325
Zabu Finance -> $16,256,815

##### Chain:
Kudex -> KCC (Kucoin)
Kswap -> Okexchain
Koffee Swap -> KCC (Kucoin)
Cycle Finance -> Avalanche
Zabu Finance -> Avalanche

##### Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list)
Cycle Finance -> "id": "cycle-token"
Zabu Finance -> "id": "zabu-token"

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
Kudex ->
Kswap -> "id":1431
Koffee Swap ->
Cycle Finance ->
Zabu Finance -> "id":11680

##### Token address and ticker if any:
Kudex Token (KUD) -> 0xBd451b952dE19F2C7bA2c8c516b0740484E953B2​
Kswap Token (OKT) -> 0xab0d1578216A545532882e420A8C61Ea07B00B12
Koffee Swap Token (KOFFEE) -> 0xc0ffee0000921eb8dd7d506d4de8d5b79b856157
Cycle Finance (CYCLE) -> 0x81440C939f2C1E34fc7048E518a637205A632a74
Zabu Finance (ZABU) -> 0xDd453dBD253fA4E5e745047d93667Ce9DA93bbCF

##### Category (Yield/Dexes/Lending/Minting) *Please choose only one:
Kudex -> Yield
Kswap -> Yield
Koffee Swap -> Dexes
Cycle Finance -> Minting
Zabu Finance -> Yield

### CODE REVIEW

**Comment by Mohammed18062020** [comment]
These protocol need of adding a complete updating of portedTokens.js file and solve the problem with coingecko as well. By now, these protocols read thier native token and that's it. The updating that I did just to run the protocol and see everthing go as I expect.

**Comment by Mohammed18062020** [comment]
Cycle Protocol is another one that is on Avalanche network.

**Comment by 0xngmi** [comment]
could you fix the merge conflicts?

**Comment by Mohammed18062020** [comment]
The merge conflicts solved!

File: projects/cyclefinance/index.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+      abi: abi.stakingToken,
+      target: coreRewards,
+      chain: "avax",
+      block: chainBlocks["avax"],
+    })
+  ).output;
+
+  const transformAddress = await transformAvaxAddress();
+
+  await sumTokensAndLPsSharedOwners(
```
Comment (0xngmi) [inline projects/cyclefinance/index.js:133]: you can merge the two calls to sumTokensAndLPsSharedOwners on a single one to make it more efficient

Comment (Mohammed18062020) [reply projects/cyclefinance/index.js:133]: yeah, but the owners are different! how we can include a condition there?

Comment (Mohammed18062020) [reply projects/cyclefinance/index.js:133]: the solucion that I have is create a function! but we are in the same thing.

Comment (Mohammed18062020) [reply projects/cyclefinance/index.js:133]: I did this and worked: ```  await sumTokensAndLPsSharedOwners(
    balances,
    [
      [staking_lpToken, true],
      [CYCLE, false],
    ],
    [coreRewards, avaxRewards],
    chainBlocks["avax"],
    "avax",
    transformAddress
  );```
But I'm not sure if it a good practic hh!

Comment (Mohammed18062020) [reply projects/cyclefinance/index.js:133]: I've got a solution for that, I'll do a PR when I can.

**Comment by Mohammed18062020** [comment]
Zabu Fiance is another protocol that is on Avalanche network!

**Pull request merged**